portal_pages/tests/test_user_forms.py

```diff
@@ -0,0 +1,137 @@
+    CaseStudyIndexPage, CaseStudyPage,
+    BlogIndexPage, BlogPage
+    )
+
+
+class UserFormTests(TestCase):
+
+    def setUp(self):
+        # Wagtail's default site comes with a Root (1) and Home Page (2)
+        self.default_site = Site.objects.get(is_default_site=True)        
+        self.root = Page.objects.get(pk=1)
+        self.home_page = Page.objects.get(pk=2)
+
+        # Add a marketplace index page under the home page
+        self.marketplace_index_page = self.home_page.add_child(
+            instance=MarketplaceIndexPage(
+                title="Marketplace",
+                slug="marketplace",
+                live=True,
+            ))
+        self.marketplace_submission_url = self.marketplace_index_page.url + self.marketplace_index_page.reverse_subpage('submit')
+
+        self.marketplace_entry_form_data = { 
+            'date_start': '2015-01-01',
+            'title': 'new marketplace',
+            'biography': '<p>bio</p>',
+            # spambuster/honeypot fields
+            'email_field_email': '',
+            'company_company_email': 'nine'
+        }
+
+        # Add a case study index page under the home page
+        self.case_study_index_page = self.home_page.add_child(
+            instance=CaseStudyIndexPage(
+                title="Stories",
+                slug="stories",
+                live=True,
+            ))
+        self.case_study_submission_url = self.case_study_index_page.url + self.case_study_index_page.reverse_subpage('submit')
+
+        self.case_study_form_data = { 
+            'date': '2015-01-01',
+            'month_start': 12,
+            'year_start': 2014,
+            'title': 'new case study',
+            'summary': '<p>summary</p>',
+            # spambuster/honeypot fields
+            'email_field_email': '',
+            'company_company_email': 'nine'
+        }
+
+        # Add a blog index page under the home page
+        self.blog_index_page = self.home_page.add_child(
+            instance=BlogIndexPage(
+                title="Blog",
+                slug="blog",
+                live=True,
+            ))
+        self.blog_submission_url = self.blog_index_page.url + self.blog_index_page.reverse_subpage('submit')
+
+        self.blog_form_data = { 
+            'date': '2015-01-01',
+            'title': 'new blog',
+            'body': '<p>body</p>',
+            'submitter_email': 'fakeone@caktusgroup.com',
+            # spambuster/honeypot fields
+            'email_field_email': '',
+            'company_company_email': 'nine'
+        }
+
+    def test_marketplace_entry_form_valid(self):
+        # This should successfully add a single new marketplace entry
+        resp = self.client.post(self.marketplace_submission_url, self.marketplace_entry_form_data)
+        self.assertEqual(resp.status_code, 302)
+        self.assertEqual(resp['Location'], 'http://testserver/%s/submit-thank-you/' % self.marketplace_index_page.slug)
```

Comment:
I'd expect to use assertRedirects here, and construct the expected value using the same method as the urls for posting:
```python
    expected_redirect = self.marketplace_index_page.url + 
        self.marketplace_index_page.reverse_subpage('thanks')
    self.assertRedirects(resp, expected_redirect)
```